src/datachain/lib/signal_schema.py

```diff
@@ -245,6 +245,9 @@ def _find_in_tree(self, path: list[str]) -> DataType:
         while curr_tree is not None and i < len(path):
             if val := curr_tree.get(path[i], None):
                 curr_type, curr_tree = val
+            elif i == 0 and (val := curr_tree.get(".".join(path), None)):
+                curr_type, curr_tree = val
```

Comment:
Won't this crash trying to unpack two values from `None` if the requested "."-based path is not found in `curr_tree` above, since the get will return `None`?

Also, `get` usually has a default of returning `None` if the requested entry is not found, providing `None` as a default value is usually not needed. See the Python documentation here: https://docs.python.org/3.12/library/stdtypes.html#dict.get

Reply:
> Won't this crash trying to unpack two values from None if the requested "."-based path is not found in curr_tree above, since the get will return None?

If I understand the question correctly I don't think either of the `if` or `elif` statements will evaluate to `True` under those circumstances because of the `:=` walrus operator. I.e. to re-assign `curr_type, curr_tree` we need `val is True`. Please correct me if I am wrong.

> Also, get usually has a default of returning None if the requested entry is not found, providing None as a default value is usually not needed. See the Python documentation here: https://docs.python.org/3.12/library/stdtypes.html#dict.get

I did this on auto-pilot, starting with the above line. I'll update both.

Reply:
Yep, that's correct - sorry I misread the `:=` operator usage (as I don't use it much). Thanks for the changes!